### DISCUSSION TITLE
importBuild path

### DISCUSSION THREAD

**Discussion by hemengke1997** [vikejs/vike#614] [Discussion] [answered]
```ts
const importPathAbsolute = toPosixPath(
    // [RELATIVE_PATH_FROM_DIST] Current file: node_modules/vite-plugin-ssr/dist/cjs/node/plugin/plugins/importBuild/index.js
    require.resolve(`../../../../../../dist/cjs/node/plugin/plugins/importBuild/loadBuild.js`)
  )
```

```ts
const importPathAbsolute  = toPosixPath(require.resolve(`./loadBuild`))
```

# Question
Why do we resolve `importBuild`'s path by a long `../`?  Is there any edge case? The dir of `loadBuild.js` is relative to `importBuild/index.js`. So maybe we can simplify it? 😁 I'm not sure, just curious.

**Comment by brillout** [comment, marked as answer]
Because we sometimes need to target `dist/esm/` instead of `dist/cjs/`, for example: https://github.com/brillout/vite-plugin-ssr/blob/09b2b967e3d1f969c2fadc52f37f6b6626f11ed0/vite-plugin-ssr/node/plugin/plugins/buildConfig.ts#L55-L56

Out of habit I always resolve starting from `dist/`. But, yea, I think for `loadBuild.js` we could indeed directly `require.resolve('./loadBuild.js')` instead. PR welcome, if you want.
